src/providers/node.rs

```diff
@@ -177,24 +182,34 @@ impl NodeProvider {
     pub fn get_nix_packages(app: &App, env: &Environment) -> Result<Vec<Pkg>> {
         let package_json: PackageJson = app.read_json("package.json")?;
         let node_pkg = NodeProvider::get_nix_node_pkg(&package_json, env)?;
-
+        let mut pkg = vec![node_pkg.clone()];
         if NodeProvider::get_package_manager(app) == "pnpm" {
             let mut pnpm_pkg = Pkg::new("nodePackages.pnpm");
             // Only override the node package if not the default one
             if node_pkg.name != *DEFAULT_NODE_PKG_NAME {
                 pnpm_pkg = pnpm_pkg.set_override("nodejs", node_pkg.name.as_str());
             }
-            Ok(vec![node_pkg, pnpm_pkg])
+            pkg.push(pnpm_pkg);
         } else if NodeProvider::get_package_manager(app) == "yarn" {
             let mut yarn_pkg = Pkg::new("yarn");
             // Only override the node package if not the default one
             if node_pkg.name != *DEFAULT_NODE_PKG_NAME {
                 yarn_pkg = yarn_pkg.set_override("nodejs", node_pkg.name.as_str());
             }
-            Ok(vec![node_pkg, yarn_pkg])
-        } else {
-            Ok(vec![node_pkg])
+            pkg.push(yarn_pkg);
         }
+        Ok(pkg)
+    }
+
+    pub fn uses_canvas(app: &App) -> bool {
+        let package_json = app.read_file("package.json").unwrap_or_default();
+        let lock_json = app.read_file("package-lock.json").unwrap_or_default();
+        let yarn_lock = app.read_file("yarn.lock").unwrap_or_default();
+        let pnpm_yaml = app.read_file("pnpm-lock.yaml").unwrap_or_default();
+        package_json.contains("canvas")
```

Comment:
I feel like this check is a bit too loose? Probably fine but it'll probably end up with quite a few false positives.

Reply:
Improved it a bit. Now, nixpacks will check for `"canvas"` in `package.json` and `/canvas/` in lockfiles.